js/listdates.js

```diff
@@ -1,87 +1,58 @@
-    { event: "Spaceninja.com", date: "1999-6-16" },
-    { event: "MacCartney, Gavin", date: "2000-4-2" },
-    { event: "Scott graduates PSU", date: "2000-6-30" },
-    { event: "Scott's first web job (RR)", date: "2001-10-11" },
-    { event: "Vandehey, Leilani", date: "2002-6-14" },
-    { event: "Scott and Annie get engaged", date: "2002-8-22" },
-    { event: "Wedding, Dave &amp; Rose", date: "2003-8-9" },
-    { event: "Wedding, Urn &amp; Kat", date: "2003-8-30" },
-    { event: "Wedding, Annie &amp; Scott", date: "2003-9-12" },
-    { event: "Stockman, Oskar", date: "2004-9-10" },
-    { event: "Stockman, Milo", date: "2007-1-26" },
-    { event: "Vandehey, Zoe", date: "2007-2-22" },
-    { event: "Stockman, Theo", date: "2009-1-27" },
-    { event: "Vandehey, John", date: "2013-3-22" },
-    { event: "Scott starts at Say Media", date: "2015-5-18" },
-    { event: "Vandehey, Gemma", date: "2016-11-7" },
-    { event: "Sasaki-Stockman, Shelby", date: "2016-11-9" },
-    { event: "Kirk, James T", date: "2233-03-22" },
-];
+var dates = [];
+var xhr = new XMLHttpRequest();
 
+// Convert date array into HTML
 var listDates = function() {
     for (var i = 0; i < dates.length; i++) {
+        var durationText = "";
+        var eventDate = moment(dates[i].date);
+        var diff = eventDate.diff(moment(), 'milliseconds');
+        var millisecondsMonth = moment.duration(1, 'months');
+
+        // use moment.js's fromNow() for durations shorter than a month
+        var shortDuration = eventDate.fromNow();
+
+        // use humanizeDuration for durations longer than a month
+        var longDuration = humanizeDuration(diff, { largest: 2, round: true });
+
+        // add human-readable duration text
+        longDuration = diff < 0 ? longDuration + " ago" : "in " + longDuration;
+
+        // decide which text to use based on how long the duration is
+        durationText = Math.abs(diff) < millisecondsMonth ?
+                       shortDuration :
+                       longDuration;
+
+        // construct the HTML table row
+        content += "<tr>" +
+                   "<td>" + dates[i].event + "</td>" +
+                   "<td class='num'>" + eventDate.calendar(null, {sameElse: 'll'}) + "</td>" +
+                   "<td class='num'>" + durationText + "</td>" +
+                   "</tr>";
+    }
+};
 
-    var durationText = "";
-    var eventDate = moment(dates[i].date);
-    var diff = eventDate.diff(moment(), 'milliseconds');
-    var millisecondsMonth = moment.duration(1, 'months');
-
-    // use moment.js's fromNow() for durations shorter than a month
-    var shortDuration = eventDate.fromNow();
-
-    // use humanizeDuration for durations longer than a month
-    var longDuration = humanizeDuration(diff, { largest: 2, round: true });
+// Handle the JSON response
+xhr.onload = function() {
+    if(xhr.status === 200) {
+        // Parse the JSON data
+        dates = JSON.parse(xhr.responseText);
```

Comment:
Is there any chance this could blow up? I think `JSON.parse` is consistently the thing that blows me out of the water more than anything else in the damn world. Like what happens if your backend starts 500ing?

I don't know if there's a better way to do exception handling than this though: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch